### PR TITLE
feat(iotda): add OTA upgrade package resource support

### PR DESCRIPTION
**Pull request by jinyangyang222**
**What this PR does / why we need it**:
add OTA upgrade package resource support
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
add OTA upgrade package resource support
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST="./huaweicloud/services/acceptance/iotda" TESTARGS="-run TestAccUpgradePackage_basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/iotda -v -run TestAccUpgradePackage_basic -timeout 360m -parallel 4
=== RUN   TestAccUpgradePackage_basic
--- PASS: TestAccUpgradePackage_basic (7.86s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/iotda     7.898s
```

### CODE REVIEW

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_upgrade_package.go [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+
+	obsLocationParam := buildObsLocationParam(raw[0].(map[string]interface{})["obs_location"])
+	fileLocationParam := model.FileLocation{
+		ObsLocation: obsLocationParam,
+	}
+
+	return &fileLocationParam
+}
+
+func buildUpgradePackageCCreateParams(d *schema.ResourceData) *model.CreateOtaPackageRequest {
```
Comment (deer-hang) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_upgrade_package.go]:
```suggestion
func buildUpgradePackageCreateParams(d *schema.ResourceData) *model.CreateOtaPackageRequest {
```

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_upgrade_package.go [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+		d.Set("description", resp.Description),
+		d.Set("custom_info", resp.CustomInfo),
+		d.Set("created_at", resp.CreateTime),
+	)
+
+	return diag.FromErr(mErr.ErrorOrNil())
+}
+
+func flattenFileLocation(resp *model.FileLocation) []interface{} {
+	obsLocation := flattenObsLocation(resp.ObsLocation)
```
Comment (deer-hang) [inline huaweicloud/services/iotda/resource_huaweicloud_iotda_upgrade_package.go:217]: For pointer types, perform null judgment in advance.
Same to function `flattenObsLocation`.

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_upgrade_package.go (hunk 1 above)
Comment (jinyangyang222) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_upgrade_package.go]: fixed.

File: huaweicloud/services/iotda/resource_huaweicloud_iotda_upgrade_package.go (hunk 2 above)
Comment (jinyangyang222) [reply huaweicloud/services/iotda/resource_huaweicloud_iotda_upgrade_package.go:217]: fixed.

**Pull request merged**